enhancements/installer/rhel8_worker_support.md

```diff
@@ -0,0 +1,252 @@
+---
+title: rhel-8-server-worker-infra-node-support
+authors:
+  - "@mtnbikenc"
+reviewers:
+  - "@staebler"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2021-05-13
+last-updated: 2021-05-13
+status: provisional
+see-also:
+  - "https://issues.redhat.com/browse/CORS-1650"
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# RHEL 8 Server Worker/Infra Node Support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes to add RHEL 8 Server support for deploying worker/infra
+nodes with [openshift-ansible] RHEL 8 nodes will be deployed and upgraded using
+the same Ansible playbook process however, RPM content will be delivered using
+an image in the release package.
+
+## Motivation
+
+Customers desire to deploy RHEL 8 nodes and using containerized content will provide
+a better user experience than configuring RPM repositories on nodes.  Using the
+existing process will allow near-term support of RHEL 8 nodes while enhanced
+longer-term RHEL 8 support is developed.
+
+### Goals
+
+1. Implement an os-content image for RHEL 8 RPMs
+1. Update [openshift-ansible] scaleup.yml and upgrade.yml playbooks to support RHEL 8 workers
+
+### Non-Goals
+
+* Deploying/upgrading RHEL 8 worker nodes through other mechanisms, i.e. operators
+* Changing the deployment/upgrade process for RHEL 7 worker nodes
+* Upgrading existing RHEL 7 worker nodes to RHEL 8
+
+## Proposal
+
+Currently, for RHEL 7 worker node deployment, the user must [prepare the host][prepare-rhel]
+before running the scaleup.yml playbook.  This process can be cumbersome for
+automating deployment and also requires additional steps to update repos prior
+to performing an upgrade.  For RHEL 8 hosts preparing repos on the hosts will
+not be required because the necessary packages will be delivered as an image in
+the release package.  This removes the need of mirroring the release package as
+well as setting up repo mirrors for offline or disconnected deployments.
```

Comment:
It's rather surprising that the text in this proposal only mentions "RHCOS" once in a later question.   Because obviously this model is a whole big advantage of what we've implemented with RHCOS today.

Also I'd really like to reinforce the story that RHCOS *is* RHEL, it's just managed by the MCO.  So we could phrase this as something more like "Support traditional RHEL8 workers, except align more with the way RHEL CoreOS works by shipping OS updates in a container" or so?

Reply:
Also there's an important flip side to this - the reason one might want to use traditional RHEL8 is *precisely* because one has existing management tools in place for it.  For example, one might be using Satellite "content views" which have existed for RHEL for a long time.  How would this model of shipping RPMs-in-containers interact with these types of external tooling?   Wouldn't these users want tooling that supported e.g. extracting the bits from this RPM and importing it to Satellite?

Reply:
Where (on what hosts) / how will a customer use this image? Will they be expected to extract it from the image and host a repo on a host (if so which one); or would this image run on the cluster and offer the RPM content to the cluster (and this is all automated)?

Reply:
I will include more verbiage about 'traditional' RHEL 8 support.

Reply:
> Where (on what hosts) / how will a customer use this image? Will they be expected to extract it from the image and host a repo on a host (if so which one); or would this image run on the cluster and offer the RPM content to the cluster (and this is all automated)?

The customer would not interact directly with the image.  The openshift-ansible playbooks would orchestrate the process of pulling the image, extracting or mounting the content, and then installing the content.